File: app/src/main/java/com/nithinkumar/water/WaterShared.java
package com.nithinkumar.water;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Created by nithinkumar on 7/21/17.
 */

public class WaterShared {

    private final String name_userValue = "name_userValue";
    private final String height_userValue = "height_userValue";
    private final String weight_userValue = "weight_userValue";
    private final String startTime_userValue = "startTime_userValue";
    private final String endTime_userValue = "endTime_userValue";
    private final String firstRun_app = "firstRun_app";
    private final String waterAdded_current = "waterAdded_current";
    private final String waterCurrentValue_daily = "waterCurrentValue_daily";
    private SharedPreferences mSharedPreferences;


    public WaterShared(Context context) {
        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public String getName_userValue() {
        return mSharedPreferences.getString(name_userValue, "UserName");
    }

    public void setName_userValue(String userName) {
        mSharedPreferences.edit().putString(name_userValue, userName).apply();
    }

    public int getHeight_userValue() {
        return mSharedPreferences.getInt(height_userValue, 00);
    }

    public void setHeight_userValue(int userHeight) {
        mSharedPreferences.edit().putInt(height_userValue, userHeight).apply();
    }

    public int getWeight_userValue() {
        return mSharedPreferences.getInt(weight_userValue, 00);
    }

    public void setWeight_userValue(int userWeight) {
        mSharedPreferences.edit().putInt(weight_userValue, userWeight).apply();
    }

    public String getStartTime_userValue() {
        return mSharedPreferences.getString(startTime_userValue, "");
    }

    public void setStartTime_userValue(String startTime) {
        mSharedPreferences.edit().putString(startTime_userValue, startTime).apply();
    }

    public String getEndTime_userValue() {
        return mSharedPreferences.getString(endTime_userValue, "");
    }

    public void setEndTime_userValue(String endTime) {
        mSharedPreferences.edit().putString(endTime_userValue, endTime).apply();
    }

    public String getFirstRun_app() {
        return mSharedPreferences.getString(firstRun_app, "");
    }

    public void setFirstRun_app(String firstRun) {
        mSharedPreferences.edit().putString(firstRun_app, firstRun).apply();
    }

    public int getWaterAdded_current() {
        return mSharedPreferences.getInt(waterAdded_current, 00);
    }

    public void setWaterAdded_current(int waterAdded) {
        mSharedPreferences.edit().putInt(waterAdded_current, waterAdded).apply();
    }

    public int getWaterCurrentValue_daily() {
        return mSharedPreferences.getInt(waterCurrentValue_daily, 00);
    }

    public void setWaterCurrentValue_daily(int waterCurrentValue) {
        mSharedPreferences.edit().putInt(waterCurrentValue_daily, waterCurrentValue).apply();
    }
}

File: app/src/main/java/com/nithinkumar/water/WaterShared.kt
package com.nithinkumar.water

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

/**
 * Created by nithinkumar on 7/21/17.
 */

class WaterShared(context: Context) {

    private val sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)

    var name: String
        get() = sharedPreferences.getString(NAME_USER_VALUE, "UserName") ?: "UserName"
        set(value) = sharedPreferences.edit().putString(NAME_USER_VALUE, value).apply()

    var height: Int
        get() = sharedPreferences.getInt(HEIGHT_USER_VALUE, 0)
        set(value) = sharedPreferences.edit().putInt(HEIGHT_USER_VALUE, value).apply()

    var weight: Int
        get() = sharedPreferences.getInt(WEIGHT_USER_VALUE, 0)
        set(value) = sharedPreferences.edit().putInt(WEIGHT_USER_VALUE, value).apply()

    var startTime: String
        get() = sharedPreferences.getString(START_TIME_USER_VALUE, "") ?: ""
        set(value) = sharedPreferences.edit().putString(START_TIME_USER_VALUE, value).apply()

    var endTime: String
        get() = sharedPreferences.getString(END_TIME_USER_VALUE, "") ?: ""
        set(value) = sharedPreferences.edit().putString(END_TIME_USER_VALUE, value).apply()

    var firstRun: String
        get() = sharedPreferences.getString(FIRST_RUN_APP, "") ?: ""
        set(value) = sharedPreferences.edit().putString(FIRST_RUN_APP, value).apply()

    var waterAddedCurrent: Int
        get() = sharedPreferences.getInt(WATER_ADDED_CURRENT, 0)
        set(value) = sharedPreferences.edit().putInt(WATER_ADDED_CURRENT, value).apply()

    var waterAddedDaily: Int
        get() = sharedPreferences.getInt(WATER_ADDED_DAILY, 0)
        set(value) = sharedPreferences.edit().putInt(WATER_ADDED_DAILY, value).apply()

    var seenOnboarding: Boolean
        get() = sharedPreferences.getBoolean(SEEN_ONBOARDING, false)
        set(value) = sharedPreferences.edit().putBoolean(SEEN_ONBOARDING, value).apply()

    companion object {
        private const val NAME_USER_VALUE = "NAME_USER_VALUE"
        private const val HEIGHT_USER_VALUE = "HEIGHT_USER_VALUE"
        private const val WEIGHT_USER_VALUE = "WEIGHT_USER_VALUE"
        private const val START_TIME_USER_VALUE = "START_TIME_USER_VALUE"
        private const val END_TIME_USER_VALUE = "END_TIME_USER_VALUE"
        private const val FIRST_RUN_APP = "FIRST_RUN_APP"
        private const val WATER_ADDED_CURRENT = "WATER_ADDED_CURRENT"
        private const val WATER_ADDED_DAILY = "WATER_ADDED_DAILY"
        private const val SEEN_ONBOARDING = "SEEN_ONBOARDING"
    }
}
